src/main/java/com/megastore/controller/CategoriesController.java

```diff
@@ -1,9 +1,33 @@
 package com.megastore.controller;
 
+import com.megastore.model.Categories;
+import com.megastore.service.CategoriesService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
 
 @RestController
 @RequestMapping("/api/v1/categories")
+@RequiredArgsConstructor
 public class CategoriesController {
+
+    private final CategoriesService categoriesService;
+
+    @GetMapping("/show")
```

Comment:
The endpoint should answer the question "What is it?". Your endpoint answered the question "What does it do?" -> show.

```suggestion
    @GetMapping("/list")
```